internal/connect/wrapper.go

```diff
@@ -0,0 +1,89 @@
+package connect
+
+import (
+	"fmt"
+
+	"github.com/SUSE/connect-ng/internal/collectors"
+	"github.com/SUSE/connect-ng/internal/credentials"
+	"github.com/SUSE/connect-ng/pkg/connection"
+	"github.com/SUSE/connect-ng/pkg/registration"
+)
+
+// Wrapper is a bridge between API connections via `pkg/connection/` and
+// `internal/credentials/`. Use this wrapper in order to perform any API
+// requests on the context of SUSEConnect.
+type Wrapper struct {
```

Comment:
What if we make Wrapper an `interface` rather then a `struct`? This would allow us to mock all `pkg/` related calls in SUSEConnect and we can just check the result there.